www/js/GenWidget.js

```diff
@@ -0,0 +1,158 @@
+/* global Widget, NNE, HTMLElement */
+/*
+  -----------
+     info
+  -----------
+
+  This Code Generator Widget extends the base Widget class to add methods
+  for more easily creating CSS generator Widgets.
+
+  -----------
+     usage
+  -----------
+
+  // this class inherits all the properties/methods of the base Widget class
+  // refer to www/js/Widget.js to see what those are
+  // or take a look at the wiki...
+  // https://github.com/netizenorg/netnet.studio/wiki/Creating-Widgets
+
+  this.codeField = this.createCodeField({
+    value: 'font-size 24px;',
+    change: (e) => this._codeFieldUpdate(e)
+  })
+
+  this.slider = w.createSlider({
+    background: '#f00',
+    min: 0,
+    max: 100,
+    label: 'PX'
+    change: (e) => this._fontSliderUpdate(e)
+  })
+
+*/
+class CodeField extends HTMLElement {
+  connectedCallback (opts) {
+    this.innerHTML = `<div>
+        <button>insert</button>
+        <input type="text">
+    </div>`
+
+    this.input = this.querySelector('input')
+    this.insertButton = this.querySelector('button')
+
+    this.querySelector('button').addEventListener('click', () => {
+      const val = this.querySelector('input').value
+      this.from = NNE.cm.getCursor('from')
+      this.to = NNE.cm.getCursor('to')
+      NNE.cm.replaceSelection(val)
+      const t = { line: this.to.line, ch: this.from.ch + val.length }
+      NNE.cm.setSelection(this.from, t)
+    })
+
+    this.querySelector('input').addEventListener('input', (e) => {
+      this.setAttribute('value', e.target.value)
+      this.change(e)
+    })
+  }
+
+  get value () {
+    return this.getAttribute('value')
+  }
+
+  set value (val) {
+    this.setAttribute('value', val)
+    const input = this.querySelector('input')
+    if (input) {
+      input.value = val
+    }
+  }
+
+  static get observedAttributes () {
+    return ['value']
+  }
+
+  attributeChangedCallback (attrName, oldVal, newVal) {
+    if (newVal !== oldVal) this[attrName] = newVal
+  }
+}
+
+window.customElements.define('code-field', CodeField)
+
+class GenWidget extends Widget {
+  constructor (opts) {
+    super(opts)
+    console.log('genwidget constructed')
+  }
```

Comment:
we don't really need that console.log, && so we don't really need this constructor at all (b/c we're not really using it for anything) so u can remove it entirely.